tfx/components/example_gen/utils.py

```diff
@@ -209,84 +213,180 @@ def _glob_to_regex(glob_pattern: Text) -> Text:
   regex_pattern = regex_pattern.replace(')', '\\)')
   return regex_pattern
 
-
-def _retrieve_latest_span(uri: Text,
-                          split: example_gen_pb2.Input.Split) -> Text:
-  """Retrieves the most recently updated span matching a given split pattern."""
+def _property_search(uri: Text,
+                     split: example_gen_pb2.Input.Split,
+                     property_names: List[Text],
+                     spec_list: List[Text]) -> List[Text]:
```

Comment:
given span/version are fundamental concepts in tfx data organization, it's unlikely we will extend to other matching groups at the moment. does it make sense to simply passing a boolean bit, e.g., is_match_version, instead of two index aligned lists? so the rest of the code can simply use SPAN_SPEC, VERSION_SPEC as constants instead of passing the two list. My concern is this looks like a premature design to handle the general case. wdyt?

Reply:
I can rewrite it to have that structure instead. my only preference for this way was (1) the previous potential extensibility (if more specs are added later, which is unlikely) and (2) less casework required in the `_property_search` method. 

It seems like we already do casework in whether span is present and whether version is present (or both), so adding this boolean seems like it produces additional casework that wasn't needed before.

However, because we're renaming this to `_glob_and_match_span_version` (more specialized function), I think this design is probably not really all that necessary. will try an implementation with a boolean.

Reply:
latest commits should have this boolean implementation